api/handlers.go

```diff
@@ -31,3 +32,123 @@ func (api *API) userHandlerGET(w http.ResponseWriter, req *http.Request, _ httpr
 	}
 	WriteJSON(w, u)
 }
+
+// userUploadsHandler returns all uploads made by the current user.
+func (api *API) userUploadsHandler(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	sub, _, _, err := tokenFromContext(req)
+	if err != nil {
+		WriteError(w, err, http.StatusInternalServerError)
+		return
+	}
+	u, err := api.staticDB.UserBySub(req.Context(), sub)
+	// TODO Should we create the user in case it's not found?
+	if err != nil {
+		WriteError(w, err, http.StatusInternalServerError)
+		return
+	}
+	ups, err := api.staticDB.UploadsByUser(req.Context(), *u)
+	if err != nil {
+		WriteError(w, err, http.StatusInternalServerError)
+	}
+	WriteJSON(w, ups)
+}
+
+// userDownloadsHandler returns all downloads made by the current user.
+func (api *API) userDownloadsHandler(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	sub, _, _, err := tokenFromContext(req)
+	if err != nil {
+		WriteError(w, err, http.StatusInternalServerError)
+		return
+	}
+	u, err := api.staticDB.UserBySub(req.Context(), sub)
+	// TODO Should we create the user in case it's not found?
+	if err != nil {
+		WriteError(w, err, http.StatusInternalServerError)
+		return
+	}
+	ups, err := api.staticDB.DownloadsByUser(req.Context(), *u)
+	if err != nil {
+		WriteError(w, err, http.StatusInternalServerError)
+	}
+	WriteJSON(w, ups)
+}
+
+// trackUploadHandler registers a new upload in the system.
+func (api *API) trackUploadHandler(w http.ResponseWriter, req *http.Request, ps httprouter.Params) {
+	sl := ps.ByName("skylink")
```

Comment:
on `trackDownload` we return a "missing parameter .." 400, we should do that here as well

Reply:
👍